File: public/dsa/arrays/inversion_count/naive/inversion_count.cpp
#include <bits/stdc++.h>
using namespace std;

int getInversionCount(vector<int> arr) {
    int inversions = 0;
    int n = arr.size();
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            if (arr[j] < arr[i]) {
                inversions++;
            }
        }
    }
    return inversions;
}

// tests-start
int main() {
    vector<int> arr = {8, 4, 1, 2};
    int inversions = getInversionCount(arr);
    cout << inversions << endl;
    return 0;
}
// tests-end

/* output
5
*/

File: public/dsa/arrays/inversion_count/naive/inversion_count.py
from typing import List


class Solution:
    def getInversionCount(self, arr: List[int]) -> int:
        inversions = 0
        n = len(arr)
        for i in range(n):
            for j in range(i + 1, n):
                if arr[j] < arr[i]:
                    inversions += 1
        return inversions


# tests-start
arr = [8, 4, 1, 2]
sol = Solution()
print(sol.getInversionCount(arr))
# tests-end

'''output
5
'''
